pandas/core/frame.py

```diff
@@ -3716,12 +3716,15 @@ def extract_unique_dtypes_from_dtypes_set(
             extracted_dtypes = [
                 unique_dtype
                 for unique_dtype in unique_dtypes
-                # error: Argument 1 to "tuple" has incompatible type
-                # "FrozenSet[Union[ExtensionDtype, str, Any, Type[str],
-                # Type[float], Type[int], Type[complex], Type[bool]]]";
-                # expected "Iterable[Union[type, Tuple[Any, ...]]]"
-                if issubclass(
-                    unique_dtype.type, tuple(dtypes_set)  # type: ignore[arg-type]
+                if (
+                    issubclass(
+                        unique_dtype.type, tuple(dtypes_set)  # type: ignore[arg-type]
+                    )
+                    or (
+                        np.number in dtypes_set
+                        and is_extension_array_dtype(unique_dtype)
+                        and unique_dtype._is_numeric
```

Comment:
use `getattr(unique_dtype, '_is_numeric', False)` as we don't actually require this on an EA type (maybe we should though)

Reply:
Done